src/components/EnterpriseContract/EnterpriseContractTable/EnterpriseContractTable.tsx

```diff
@@ -43,20 +44,56 @@ export const getSortColumnFuntion = (key: string, activeSortDirection: string) =
   }
 };
 
+// Add Reat.memo to avoid unexpected redender when the ecResult has no change.
 export const EnterpriseContractTable: React.FC<
   React.PropsWithChildren<EnterpriseContractTableProps>
-> = ({ ecResult }) => {
+> = React.memo(({ ecResult }) => {
   const [activeSortIndex, setActiveSortIndex] = React.useState<number | null>(2);
   const [activeSortDirection, setActiveSortDirection] = React.useState<'asc' | 'desc' | null>(
     'asc',
   );
+  // We add pagination support here to avoid large ecResult.
+  // When one application has more than 100 components, the size of ecResult is more than
+  // 23M and the length of records is more than 50000. The browser cannot show it in one page.
+  // More, we cannot enjoy the shared virtualized table for inconsisitent parameters.
```

Comment:
@sahil143 , i have tried the visualized table when enjoyed by pipelinerun and others but failed for inconsistent parameters. would you like to double-confirm?

Reply:
What do you mean by inconsistent parameters here?

Reply:
The visualized table is supporting the infinite scroll scenarios like the pipelineruns and other lists.
For this log parts, it does not match that scenario.

Reply:
I would try to enjoy the visualized table, thank you for pointing it out. :-)